src/ol/renderer/canvas/canvasvectorlayerrenderer.js

```diff
@@ -194,6 +227,14 @@ ol.renderer.canvas.VectorLayer.prototype.prepareFrame =
 
   var extent = ol.extent.buffer(frameStateExtent,
       vectorLayerRenderBuffer * resolution);
+  var projectionExtent = viewState.projection.getExtent();
+
+  if (vectorSource.getWrapX() && viewState.projection.isGlobal() &&
```

Comment:
As far as I know, isGlobal is indeed only used for wrapX. The decision whether to wrap the x-axis is a 2-step process though. First of all, you need to know the extent of the world, because the extent edges are where you "connect" a wrapped world. And then it's up to the user to decide whether they want to wrap the x-axis. In OpenLayers 2, we left the whole decision to the user, and often got questions why a layer is repeated at its extent boundaries.